### DISCUSSION TITLE
Randomize Directed Graphs

### DISCUSSION THREAD

**Discussion by M-Schurr** [networkx/networkx#6206] [Ideas]
In networkx there are different approaches to randomize existing graphs (e.g. _random_reference_). The best known approach is to swap the edges of the graph, which preserves the degree distribution. 

The idea of my study group would be to extend the above function so that this is also possible for directed graphs. 

Specifically, we would refer to this paper: https://journals.aps.org/pre/pdf/10.1103/PhysRevE.85.046103?casa_token=-uyLVPELX6oAAAAA%3AFXQDycO9obyO7D0h1Z4tbKNUeMF80VZ9IXmI14Odn3KxydfjRcdR6AIaz6oI90aCp_12ZP9pHALwxX1Q

My question would be if there is already a way to randomize directed graphs in networkx and if not, was this intentionally not added or is this addition simply missing in the package?

Thanks for the answer

**Comment by dschult** [comment]
I guess a good place to look is: [directed_edge_swap](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.swap.directed_edge_swap.html)
I'm not sure if that is what you are looking for or not. But it's something. :)

**Reply by M-Schurr** [reply]
I already know this method. However, there are many different theoretical considerations to perform a randomization of directed graphs. Have any efforts been made to implement a different method?
Thank you for your answer.

**Reply by dschult** [reply]
I'm confused as to why you knew about this function but didn't mention it in your original post...???
We are open to new methods for edge swapping with directed graphs. I would suggest thought that you not simply rely on a single paper for your study group. Explore the literature and see if you can figure out which method have staying power -- that is, are suggested in one paper and then have other papers building on that idea. Those are probably the best choices for inclusion. It makes the study group more involved in literature searches than jumping into code, but I think that is a helpful activity for the group too. :)  just my 2 cents...  you do what is best for the study group...
